stravalib/model.py

```diff
@@ -116,6 +125,80 @@ def __repr__(self):
         )
 
 
+class DeprecatedSerializableMixin(BaseModel):
+    """
+    Provides backward compatibility with legacy BaseEntity
+    """
+
+    @classmethod
+    def deserialize(cls, v):
+        """
+        Creates and returns a new object based on serialized (dict) struct.
+        """
+        warn_method_deprecation(
+            cls,
+            "deserialize()",
+            "parse_obj()",
+            "https://docs.pydantic.dev/usage/models/#helper-functions",
+        )
+        return cls.parse_obj(v)
+
+    def from_dict(self, v):
+        """
+        Deserializes v into self, resetting and ond/or overwriting existing fields
+        """
+        warn_method_deprecation(
+            self.__class__.__name__,
+            "from_dict()",
+            "parse_obj()",
+            "https://docs.pydantic.dev/usage/models/#helper-functions",
+        )
+        # Ugly hack is necessary because parse_obj does not behave in-place but returns a new object
+        self.__init__(**self.parse_obj(v).dict())
+
+    def to_dict(self):
+        """
+        Returns a dict representation of self
+        """
+        warn_method_deprecation(
+            self.__class__.__name__,
+            "to_dict()",
+            "dict()",
+            "https://docs.pydantic.dev/usage/exporting_models/",
+        )
+        return self.dict()
+
+
+class BackwardCompatibilityMixin:
+    """
+    Mixin that intercepts attribute lookup and raises warnings or modifies return values
+    based on what is defined in the following class attributes:
+    * _deprecated_fields (TODO)
+    * _unsupported_fields (TODO)
+    * _field_conversions
+    * _unit_registry
+    """
+
+    def __getattribute__(self, attr):
+        value = object.__getattribute__(self, attr)
+        if attr in ["_field_conversions", "_unit_registry"]:
+            return value
+        try:
+            if attr in self._field_conversions:
+                value = self._field_conversions[attr](value)
```

Comment:
`_field_conversions` is a class attribute that maps field names to operations that change their type/value when requested

Reply:
ok last question and i think i will have enough to be a bit more helpful in testing. i know this is a private attribute. but what is an example of how this is used? 

is this a helper related to unit conversion meters to feet, etc? many thanks.

Reply:
`_field_conversions` are used when an attribute's value from the pydantic model doesn't match the type or format of the legacy attribute. For example, the Strava API (and pydantic) models durations as simple integers representing seconds. However, the legacy API uses `timedelta` objects for durations. To preserve this behavior, we must convert certain attributes from the pydantic model. The `ActivityTotals` class gives an example of how this is used.

Reply:
ok I will play with this a bit more. I totally remember that timedelta duration as it actually messed up activities that were >24 hours (100 milers) and it took me a minute to figure out that is why my data were off.